cmd/commands/server/factory.go

```diff
@@ -58,6 +59,10 @@ func NewCommandWith(
 		locationDetector = location.NewDetectorFake("")
 	}
 
+	// (Re)generate required security primitives before openvpn start
+	openVPNPrimitives := primitives.NewOpenVPNSecPrimitives()
+	openVPNPrimitives.Generate()
```

Comment:
What happens if some certificate generation failed? When can this happen?

Reply:
Is it really necessary to have methods which modify state? Maybe we can have more functional style like primitives := GeneratePrimitives()

Reply:
@tadovas , you might want to generate just a few of all primitives. Renamed Generate() to GenerateAll()